app/ante/ante.go

```diff
@@ -53,13 +54,30 @@ func (r AdjustGasPriceDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 		ctx = ctx.WithMinGasPrices(sdk.NewDecCoins(minGasPrice))
 		return next(ctx, tx, simulate)
 	}
-	feeTx, ok := tx.(sdk.FeeTx)
-	if !ok {
-		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
-	}
-	fees := feeTx.GetFee()
-	if len(fees) == 1 && fees[0].Amount.LT(sdk.NewInt(100000000000000000)) {
-		return ctx, sdkerrors.Wrap(sdkerrors.ErrLogic, "fee is too low")
+	for i := range msgs {
+		if msgs[i].Type() == banktypes.TypeMsgSend || msgs[i].Type() == banktypes.TypeMsgMultiSend ||
+			msgs[i].Type() == "createUserClaim" || msgs[i].Type() == "swap" ||
+			msgs[i].Type() == "remove_liquidity" || msgs[i].Type() == "add_liquidity" {
+			feeTx, ok := tx.(sdk.FeeTx)
+			if !ok {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+			}
+			fees := feeTx.GetFee()
+			if len(fees) == 1 && fees[0].Amount.LT(sdk.NewInt(100000000000000000)) { // 0.1
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrLogic, "fee is too low")
+			}
+			return next(ctx, tx, simulate)
+		} else if msgs[i].Type() == "transfer" {
```

Comment:
@iverc Do we want to have a catch-all `else` just in case we add a new message type in the future?

Reply:
@intl-man we do not enforce min fee for all other messages, if we add a new message type it should be explicitly included in the if check to enforce min fee, otherwise we can skip it

Reply:
@iverc Gotcha. Just checking. Looks good!